### ISSUE TITLE
cgen: fixed-size array initialisation with `it`

### ISSUE THREAD

**Issue by Hunam6** [vlang/v#14237]
```
OS: macos, macOS, 12.3.1, 21E258
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2)

getwd: /Users/hunam/Documents/dev/go2v
vmodules: /Users/hunam/.vmodules
vroot: /Applications/v
vexe: /Applications/v/v
vexe mtime: 2022-04-30 21:32:06
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 8cc79e4.ec92d46

Git version: git version 2.32.0 (Apple Git-132)
Git vroot status: weekly.2022.16-115-gec92d467
.git/config present: true
thirdparty/tcc status: thirdparty-unknown-unknown de82a130
```

**What did you do?**
```v
fn main() {
	arr := [1]int{init: it}
}
```

**What did you expect to see?**
`arr` equals `[0]`
 
**What did you see instead?**
```c
/tmp/v_501/a.9052813162094903872.tmp.c:2054:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/a.9052813162094903872.tmp.c:11418:20: error: array initializer must be an initializer list or wide string literal
        Array_fixed_int_1 arr =_t1;
                          ^
1 warning and 1 error generated.
```

**Comment by depperm** [comment]
I think the correct syntax is `arr := []int{len: 1, init: it}`

**Comment by JalonSolov** [comment]
That is correct for a regular array.  He's trying to create a fixed array.  What he has should probably work, or the simpler `arr := [0]!` can be used (note the trailing `!`).

**Comment by Hunam6** [comment]
I'm indeed trying to initiate a fixed array, I'm not using the simpler syntax because I want to be able to put some more complex length like `1 << 9`, like so: `[1 << 9]int{init: it}`.

**Comment by DevrosTheOne** [comment]
something like?
```v
arr := []int{len: 9, init: it + 1}
[1, 2, 3, 4, 5, 6, 7, 8, 9]
```

**Comment by Hunam6** [comment]
This isn't the same program at all, in my program the array has a fixed-size and 512 elements (`1 << 9` == `512`) but whatever, the example I linked **should** work.

**Comment by DevrosTheOne** [comment]
sorry I misread/understood the code 😵

**Comment by JalonSolov** [comment]
This was fixed a few minutes ago by #14251

**Comment by JalonSolov** [comment]
This is similar to one you just fixed a few minutes ago, yuyi

**Comment by JalonSolov** [comment]
Never mind.  It was fixed by #14248

**Comment by yuyi98** [comment]
It is same as #14248.
